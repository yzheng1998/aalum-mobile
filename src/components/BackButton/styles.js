import styled from 'styled-components'

export const IconWrapper = styled.TouchableOpacity`
  position: absolute;
  margin-top: 30px;
  margin-left: 20px;
`
