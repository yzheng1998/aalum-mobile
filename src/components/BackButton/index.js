import React, { Component } from 'react'
import Icon from '../Icon'
import { IconWrapper } from './styles'

export default class BackButton extends Component {
  render() {
    return (
      <IconWrapper onPress={this.props.onPress}>
        <Icon name="arrow-back" color="black" size={30} />
      </IconWrapper>
    )
  }
}
