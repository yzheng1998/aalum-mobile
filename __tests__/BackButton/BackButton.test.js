import React from 'react'
import BackButton from '../../src/components/BackButton'
import { shallow } from 'enzyme'

describe('BackButton', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<BackButton />)
    expect(wrapper).toMatchSnapshot()
  })

  test('handles onPress correctly', () => {
    const onPressMock = jest.fn()
    const wrapper = shallow(<BackButton onPress={onPressMock} />)
    const touchable = wrapper.find('Button')
    touchable.simulate('press')

    expect(onPressMock).toHaveBeenCalledTimes(1)
  })
})
