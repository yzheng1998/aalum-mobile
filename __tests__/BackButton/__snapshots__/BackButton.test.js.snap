// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly 1`] = `
<Button
  hitSlop={
    Object {
      "bottom": 5,
      "left": 5,
      "right": 5,
      "top": 5,
    }
  }
>
  <_default
    name="arrow-back"
    size={30}
  />
</Button>
`;
